packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -98,13 +98,11 @@ async function ssrTransformScript(
   // hoist at the start of the file, after the hashbang
   const hoistIndex = code.match(hashbangRE)?.[0].length ?? 0
 
-  function defineImport(source: string) {
+  function defineImport(index = 0, source: string) {
```

Comment:
```suggestion
  function defineImport(index: number, source: string) {
```

Would be great to make `index` stricter